gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonElementConversionWithAvroSchemaFactory.java

```diff
@@ -210,4 +215,49 @@ public Schema schema() {
       return this.schema;
     }
   }
+
+  public static class UnionConverter extends ComplexConverter {
+    private final Schema firstSchema;
+    private final Schema secondSchema;
+    private final JsonElementConverter firstConverter;
+    private final JsonElementConverter secondConverter;
+
+    public UnionConverter(String fieldName, boolean nullable, String sourceType, Schema schemaNode,
+        WorkUnitState state, List<String> ignoreFields) throws UnsupportedDateTypeException {
+      super(fieldName, nullable, sourceType);
+      List<Schema> types = schemaNode.getTypes();
+      if (types.size() != 2) {
+        throw new RuntimeException("Unions must be size 2.");
```

Comment:
This is not a valid assumption in the context of Avro schema. What's the reason to enforce `size == 2` for union here?

Reply:
the existing code only supports two types.
https://github.com/apache/incubator-gobblin/blob/master/gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonRecordAvroSchemaToAvroConverter.java#L93